scale_build/utils/run.py

```diff
@@ -0,0 +1,44 @@
+import os
+import pexpect
+import subprocess
+
+from scale_build.exceptions import CallError
+
+
+def run(*args, **kwargs):
+    if isinstance(args[0], list):
+        args = tuple(args[0])
+    kwargs.setdefault('stdout', subprocess.PIPE)
+    kwargs.setdefault('stderr', subprocess.PIPE)
+    exception_message = kwargs.pop('exception_msg', None)
+    check = kwargs.pop('check', True)
+    shell = kwargs.pop('shell', False)
+    logger = kwargs.pop('logger', None)
+    env = kwargs.pop('env', None) or os.environ
+    if logger:
+        kwargs['stderr'] = subprocess.STDOUT
+
+    proc = subprocess.Popen(args, stdout=kwargs['stdout'], stderr=kwargs['stderr'], shell=shell, env=env)
+    if logger:
+        for line in map(lambda l: l.rstrip().decode(errors='ignore'), iter(proc.stdout.readline, b'')):
```

Comment:
Is `stderr` ignored?

Reply:
And can't you just open `stdout` and `stderr` with `encoding` and `errors` by default?

Reply:
It's not ignored
```
    if logger:
        kwargs['stderr'] = subprocess.STDOUT
```

Reply:
Although i think i should separate logger implementation from the default `run` function. Thank you for pointing it out.

Reply:
You are not consuming it here, it could lead to a deadlock.

Reply:
Sorry, totally missed that part.

Popening with `encoding` and `errors` by default is still a suggestion.